applications/suit/flash_companion/README.rst

```diff
@@ -0,0 +1,84 @@
+.. _suit_flash_companion:
+
+SUIT flash companion
+####################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The flash companion application allows the secure domain firmware to access the external memory during the SUIT firmware upgrade.
+
+Application overview
+********************
+
+The flash companion implements a device driver for the external flash device and provides an IPC service to the secure domain.
+The secure domain firmware can request from the flash companion to read, erase, or write data to the external memory.
+
+The application is meant to be booted by the secure domain while performing the firmware update process using the SUIT firmware upgrade.
```

Comment:
Link to SUIT docs missing.

Reply:
Added link to the documentation about SUIT.